enigma-tools-u/src/common_u/errors.rs

```diff
@@ -1,32 +1,34 @@
+use thiserror::Error;
+
 use sgx_types::sgx_status_t;
 
 // error while request attestation service
-#[derive(Fail, Debug)]
-#[fail(display = "Error while using the attestation service info = ({})", message)]
+#[derive(Error, Debug)]
+#[error("Error while using the attestation service info = ({message:?})")]
```

Comment:
I'm not sure about the pros/cons here in terms of readability, but strictly speaking adding all the `:?` here is a change from the original declaration, as it will use the `Debug` implementation of the fields instead of their `Display` implementation.